hutch_python/epics_arch.py

```diff
@@ -93,13 +116,138 @@ def create_arch_file(experiment, hutch=None, path=None, dry_run=False):
             file_path = path
         elif hutch:
             file_path = EPICS_ARCH_FILE_PATH.format(hutch.lower())
+        elif cds_items:
+            pull_cds_data(experiment, cds_items)
+            return
+        elif link:
+            update_file(exp_name=experiment, path=EPICS_ARCH_FILE_PATH.format(experiment[0:3]))
+            create_softlink(experiment)
+            return
         else:
             file_path = EPICS_ARCH_FILE_PATH.format(experiment[0:3])
```

Comment:
The sequencing in this code block is problematic.
- `cds_items` and `link` will not be checked if `path` or `hutch` is provided, since this is a big if/elif tree.
- In general, this if/elif tree originally was used to determine the filepath to use, so it's not a very convenient place to check unrelated arguments
- the `link` argument ignores all pre-existing `EPICS_ARCH_FILE_PATH` handling and does its own thing here
- it's not clear from the documentation that "cds_items" is not supposed to update the file and "link" is still supposed to update the file

You probably want to check `cds_items` in a separate if block, prior to checking for path, and you probably want to check `link` after the file is updated in the normal space below (to simply add the softlink step). I think the help strings also need to be more clear that `--link` still runs update but `--cds-items` does not.